### ISSUE TITLE
BasicText did not recalculate the width correctly.

### ISSUE THREAD

**Issue by notbetop** [JetBrains/compose-multiplatform#3434]
**Describe the bug**

After moving from screens with different resolutions, the width of BasicText remains unchanged.

**Affected platforms**
Select one of the platforms below:
- Desktop

**Versions**
- Kotlin version*:  1.9.0
- Compose Multiplatform version*:  .1.5.0-beta01
- OS version(s)* (required for Desktop and iOS issues): Windows11
- OS architecture (x86 or arm64):  x86
- JDK (for desktop issues): OpenJDK 17 JBR17 

**To Reproduce**

STEP1-code like this

``` kotlin
fun main() {
    singleWindowApplication {
        BasicText("Hello World!", modifier = Modifier.background(Color.Red))
    }
}


```

STEP2 - run compose
in my 2k monitor (main monitor)

![image](https://github.com/JetBrains/compose-multiplatform/assets/128110965/ca21f09b-1105-4d3b-9cf8-ad3551648d8e)

STEP3-move window to another screen(4k monitor)

you will see error display with basic text.

![image](https://github.com/JetBrains/compose-multiplatform/assets/128110965/4231bdaf-d50b-40c9-9716-a42724ef7a0c)

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by Walingar** [comment]
Reproduced on MacOs as well 
![image](https://github.com/JetBrains/compose-multiplatform/assets/23081446/4639890f-53f1-4a89-9a22-e4e8b2c22e02)

**Comment by Walingar** [comment]
Checked on:
```
kotlin.version=1.8.0
agp.version=7.3.0
compose.version=1.4.3
```

Works fine, so this is 1.5.0 regression

**Comment by Walingar** [comment]
Is going to be fixed in 1.5.0
Related commit: https://github.com/JetBrains/compose-multiplatform-core/commit/882521f410f4f9d7854b4ca5ca54277d22556012

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3434">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
